feat(RELEASE-981): refactor verify-access-to-resources into a standalone task

seanconroy2021 opened the pull request:
feat([RELEASE-981](https://issues.redhat.com//browse/RELEASE-981)): refactor verify-access-to-resources into a standalone task

This PR refactors the `verify-access-to-resources` script from being hard coded in the pipeline to a dedicated Tekton task.

**Changes:**
- Refactored pipelines to utilize the new `verify-access-to-resources` task.
- Removed the inline script from each pipeline and replaced it with a task reference.
- Updated version and documentation to reflect these changes.
- **Task details:**
  - Added a README for the new task.
  - Implemented a test to ensure the task correctly verifies resource accessibility.

**Note:** 
_verifying the no-access scenario test for the task would require creating a new PipelineRun and RoleBindings, followed by updating `test_tekton_tasks.sh`. Although possible, this approach is overly complex. The task has been validated to work as expected because the command `kubectl auth can-i get releaseplanadmission -n <namespace>  --as=system:serviceaccount:default:default` correctly returns "no." Thus, we can trust this verification. Additionally, a test is already in place to check if the resources are accessible. If the task cannot access the resources, the task will exit with `exit 1`._

Signed-off-by: Sean Conroy <sconroy@redhat.com>

seanconroy2021 committed:
feat(RELEASE-981): refactor verify-access-to-resources into a task

This PR refactors the `verify-access-to-resources` script from being
hard-coded in the pipeline to a dedicated Tekton task.

**Changes:**
- Refactored pipelines to use the new `verify-access-to-resources` task.
- Removed the inline script from each pipeline and replaced it with a
  task reference.
- Updated version and documentation to reflect these changes.

**Task details:**
- Added a README for the new task.
- Implemented a test to ensure the task correctly verifies resource
  accessibility.

Signed-off-by: Sean Conroy <sconroy@redhat.com>

seanconroy2021 commented:
/test all

johnbieren commented on application-api:
this should not be part of the PR

johnbieren commented on pipelines/rh-advisories/rh-advisories.yaml:
```diff
@@ -4,7 +4,7 @@ kind: Pipeline
   name: rh-advisories
   labels:
-    app.kubernetes.io/version: "0.11.1"
+    app.kubernetes.io/version: "0.12.1"
```
this should be 0.12.0

johnbieren replied:
You updated the tekton minVersion instead of the pipeline version after my comment

seanconroy2021 replied:
Changed it now 👍🏻

johnbieren commented on tasks/verify-access-to-resources/README.md:
```diff
@@ -0,0 +1,13 @@
+
+| Name                 | Description                                        | Optional | Default value |
+|----------------------|----------------------------------------------------|----------|---------------|
+| release              | Namespace/name of the Release                     | No       | -             |
```
The spacing is off from here down (needs one more space in the description)

johnbieren commented on tasks/verify-access-to-resources/README.md:
```diff
@@ -0,0 +1,13 @@
+| releasePlan          | Namespace/name of the ReleasePlan                 | No       | -             |
+| releasePlanAdmission | Namespace/name of the ReleasePlanAdmission        | No       | -             |
+| releaseServiceConfig | Namespace/name of the ReleaseServiceConfig        | No       | -             |
+| snapshot             | Namespace/name of the Snapshot                    | No       |
```
missing final |

johnbieren commented on tasks/verify-access-to-resources/verify-access-to-resources.yaml:
```diff
@@ -0,0 +1,81 @@
+  params:
+    - name: release
+      description:
+        The namespaced name (namespace/name) of the Release custom resource
```
These descriptions should match what you have in the README (I would vote to copy the ones from there to here as the capitalization of the resources is not consistent in the descriptions in this file)

johnbieren commented on tasks/verify-access-to-resources/verify-access-to-resources.yaml:
```diff
@@ -0,0 +1,81 @@
+        The namespaced name (namespace/name) of the Release custom resource
+      type: string
+    - name: releasePlan
+      description: 
```
If the line doesn't go over (I think it's 120) the character limit, you can have the text on the same line

johnbieren commented on tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml:
```diff
@@ -0,0 +1,96 @@
+                name: releaseplanadmission-sample
+                namespace: default
+              EOF
+              kubectl apply -f releaseplanadmission
```
I am surprised this works. There are required fields in the spec, but I guess since you omitted the spec struct entirely it is okay. Nice

johnbieren commented on tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml:
```diff
@@ -0,0 +1,96 @@
+                namespace: default
+              EOF
+              kubectl apply -f snapshot
+
```
unneeded new line

johnbieren commented on release-service:
this should not be here

johnbieren commented on internal-services:
this should not be part of the pr

seanconroy2021 commented:
/test all

seanconroy2021 commented:
Thank you for the review, @johnbieren. I don't know how the internal services, release services, and application API got into the PR.

seanconroy2021 commented:
/test release-pipelines-e2e-suite

seanconroy2021 commented:
/test all

seanconroy2021 commented:
/retest

johnbieren commented:
Can you squash these all down to one commit? You are failing gitlint

johnbieren reviewed (commented):
Just the one version thing to fix and squashing the commits then I will approve. One thing to note - typically we let the reviewer resolve their comment once the change is made and they are satisfied, instead of the PR submitter resolving their comments (to avoid situations like the rh-advisories version where the thing wasn't actually fixed)

seanconroy2021 commented:
> Just the one version thing to fix and squashing the commits then I will approve. One thing to note - typically we let the reviewer resolve their comment once the change is made and they are satisfied, instead of the PR submitter resolving their comments (to avoid situations like the rh-advisories version where the thing wasn't actually fixed)

ooh apologize for resolving the comments myself. I wasn't aware that it was handled that way . I've squashed the commits and fixed the issue with the version. It should be ready to go :)

johnbieren reviewed (approved):
/retest
